+Base/pref.m

```diff
@@ -14,6 +14,7 @@
     %       help_text - provide longer text to describe what this pref does (think tooltip)
     %       readonly - boolean specifying if GUI control should be editable (e.g. "enabled")
     %       display_only - boolean specifying if saved as pref when module unloaded
+    %       tag - any data type available for user to choose (similar to gobject UserData)
```

Comment:
Why is this necessary?

Reply:
I think it can be useful to store any additional data to make it easier on callbacks. I would push to call it UserData instead since that is literally what @ponagehl2  compare it to.

Reply:
So the `set` methods should all be `'setNum'`? Currently `set == tag` for each one. I don't really like this solution, especially because if you want to be pedantic, you would have to provide tags or a system to interpret tags for each function (clean, validate, etc). Perhaps a more elegant solution is to pass a cell array `{'setNum', 'setPower'}` and then have `pref_handler` parse and insert it upon function handle construction. I also don't think it's terrible to have a separate function for each one.

Reply:
I don’t disagree, but having this option can be quite handy. Not the worst to have a single set method for many properties. Perhaps the usage of it can be improved in this context, but I do like the added freedom it gives developers.